### PR TITLE
Added Mermaid and SVG for Common Interactions Sequence Diagram

### PR DESCRIPTION
**Pull request by midnightconman**
/kind documentation

This PR adds a definition file (in mermaid) for a simple and common interactions flow of an example pull request submission. Please let me know what you guys think.

/cc @stevekuznetsov @fejta

### CODE REVIEW

**Comment by midnightconman** [comment]
Is there a way to force `docs-no-retest`, this PR doesn't need to be tested in my opinion.

File: prow/README.md [hunk 1]
```diff
@@ -8,6 +8,8 @@ not make any attempt to preserve backwards compatibility.
 
 For a brief overview of how Prow runs jobs take a look at ["Life of a Prow Job"](/prow/life_of_a_prow_job.md).
 
+For a sequence diagram of common usage and interactions flow, click [here](/prow/docs/pr-interactions-sequence.svg).
```
Comment (fejta) [inline prow/README.md]: https://github.com/kubernetes/test-infra/blob/d7eadcf7f89bd227b5175a356e413e8bebf3fce0/prow/docs/pr-interactions-sequence.svg looks kind of small?

Comment (midnightconman) [reply prow/README.md]: Right click on it and select something like "open in a new tab", the SVG should scale. You should also be able to zoom in as far as you like. :smile:

Comment (midnightconman) [reply prow/README.md]: I see what you mean... let me try a link to the raw with sanitize on.

Comment (midnightconman) [reply prow/README.md]: I updated the link to raw for master, to see what it looks like on this commitish, use this link: https://raw.githubusercontent.com/kubernetes/test-infra/d7eadcf7f89bd227b5175a356e413e8bebf3fce0/prow/docs/pr-interactions-sequence.svg?sanitize=true

**Pull request merged**